vispy/scene/cameras/cameras.py

```diff
@@ -71,43 +197,191 @@ def interactive(self):
+        during resetting, window resizing, and interaction.
+        """
+        return self._aspect_fixed
+    
+    @aspect_fixed.setter
+    def aspect_fixed(self, value):
+        self._aspect_fixed = bool(value)
+    
+    @property
+    def center(self):
+        """ The center location for this camera
+        
+        The exact meaning of this value differs per type of camera, but
+        generally means the point of interest or the rotation point.
+        """
+        return self._center_loc or (0, 0, 0)
+    
+    @center.setter
+    def center(self, val):
+        if len(val) == 2:
+            self._center_loc = float(val[0]), float(val[1]), 0.0
+        elif len(val) == 3:
+            self._center_loc = float(val[0]), float(val[1]), float(val[2])
+        else:
+            raise ValueError('Center must be a 2 or 3 element tuple')
+        self.view_changed()
+    
+    @property
+    def fov(self):
+        """ Field-of-view angle of the camera. If 0, the camera is in
+        orthographic mode.
+        """
+        return self._fov
+    
+    @fov.setter
+    def fov(self, fov):
+        fov = float(fov)
+        if fov < 0 or fov >= 180:
+            raise ValueError("fov must be between 0 and 180.")
+        self._fov = fov
+        self.view_changed()
+    
+    def set_range(self, xrange=None, yrange=None, zrange=None, margin=0.05):
+        """ Set the range of the view region for the camera
+        
+        The view is reset to the given range or to the scene boundaries
+        if ranges are not specified. The ranges should be 2-element
+        tuples specifying the min and max for each dimension.
+        """
+        # Do we have a viewbox?
+        if self._viewbox is None:
+            return
+        self._resetting = True
+        # Size factor
+        size = self._viewbox.size
+        self._window_size_factor = size[1] / size[0]
+        # Get bounds
+        bounds = []
+        bounds_scene = self._viewbox.get_scene_bounds()
+        for i, lim in enumerate((xrange, yrange, zrange)):
+            if lim is None:
+                bounds.append(bounds_scene[i])
+            else:
+                bounds.append((float(lim[0]), float(lim[1])))
+        # Calculate ranges and margins
+        ranges = [b[1] - b[0] for b in bounds]
+        margins = [(r*margin or 0.1) for r in ranges]
+        # Assign limits for this camera
+        bounds_margins = [(b[0]-m, b[1]+m) for b, m in zip(bounds, margins)]
+        self._xlim, self._ylim, self._zlim = bounds_margins
+        # Store center location
+        self._center_loc = [(b[0] + r / 2) for b, r in zip(bounds, ranges)]
+        # Set given params (since _set_range() might derive other values)
+        for name, value in self._given_params.items():
+            setattr(self, name, value)
+        # Let specific camera handle it
+        self._set_range()
+        # Overwrite given params
+        for name, value in self._given_params.items():
```

Comment:
It doesn't look like there should be any memory of the "initial state". When we call set_range, the result should just be a combination of the arguments given and the current state of the camera; no need to refer back to the initial state.